vernac/comDefinition.ml

```diff
@@ -101,6 +101,7 @@ let do_definition ~program_mode ?hook ~name ~scope ~poly ~kind univdecl bl red_o
     ignore(Obligations.add_definition
              ~name ~term:c cty ctx ~univdecl ~implicits:imps ~scope ~poly ~kind ?hook obls)
   else
+    let kind = Decls.IsDefinition kind in
```

Comment:
Can you explain this? It seems do_definition takes in input ~kind, but *in this branch* its value is ignored/overridden.

Reply:
never mind, I now see where the discrepancy comes from... and is not this PR

Reply:
Indeed the handling of the whole situation with kinds and declare definitions vs lemmas is quite messy, I dunno how to proceed, there are like 4 / 5 tasks to do but I'm not sure what is the right order is.

Reply:
The right thing to do is to make the `kind` parameter more general, so we can also call this code from `Lemmas`.